state-chain/custom-rpc/src/lp.rs

```diff
@@ -413,8 +413,14 @@ where
 		tick: Option<Tick>,
 		sell_amount: NumberOrHex,
 		dispatch_at: Option<BlockNumber>,
+		expire_at: Option<BlockNumber>,
 		wait_for: Option<WaitFor>,
 	) -> RpcResult<ApiWaitForResult<Vec<LimitOrder>>> {
+		if let (Some(dispatch_at), Some(expire_at)) = (dispatch_at, expire_at) {
+			if dispatch_at >= expire_at {
+				Err(anyhow!("Invalid expire_at, must be larger than dispatch_at"))?;
+			}
+		};
```

Comment:
Putting this check in the runtime would also avoid having to duplicated it here..